rpcs3/Emu/Cell/SPUCommonRecompiler.cpp

```diff
@@ -1264,7 +1264,22 @@ bool spu_program::operator<(const spu_program& rhs) const noexcept
 	// Select range for comparison
 	std::span<const u32> lhs_data(data.data() + lhs_offs, data.size() - lhs_offs);
 	std::span<const u32> rhs_data(rhs.data.data() + rhs_offs, rhs.data.size() - rhs_offs);
-	const auto cmp0 = std::memcmp(lhs_data.data(), rhs_data.data(), std::min(lhs_data.size_bytes(), rhs_data.size_bytes()));
+
+	int cmp0 = 0;
+
+	if (lhs_data.size() != rhs_data.size())
+	{
+		cmp0 = std::memcmp(lhs_data.data(), rhs_data.data(), std::min(lhs_data.size_bytes(), rhs_data.size_bytes()));
+
+		if (cmp0 == 0 && lhs_data.size() < rhs_data.size())
+		{
+			cmp0 = lhs_data.size() < rhs_data.size() ? -1 : 1;
```

Comment:
Isn't result of `lhs_data.size() < rhs_data.size()` already known at this point?

Reply:
Fixed

Reply:
The same issue is on line 45 now

Reply:
Its outdated

Reply:
Ah yeah, I was one force push behind